test/e2e/operator_test.go

```diff
@@ -573,6 +573,167 @@ func fetchRouterCanonicalHostname(t *testing.T, routeName types.NamespacedName,
+	credSecret := makeWrongCredentialsSecret(operatorNamespace)
+	err = kubeClient.Create(context.TODO(), credSecret)
+	if err != nil {
+		t.Fatalf("Failed to create credentials secret %s/%s: %v", credSecret.Namespace, credSecret.Name, err)
+	}
+
+	t.Log("Creating external dns instance")
+	extDNS := helper.buildExternalDNS(testExtDNSName, hostedZoneID, hostedZoneDomain, credSecret)
+	if err := kubeClient.Create(context.TODO(), &extDNS); err != nil {
+		t.Fatalf("Failed to create external DNS %q: %v", testExtDNSName, err)
+	}
+	defer func() {
+		_ = kubeClient.Delete(context.TODO(), &extDNS)
+	}()
+
+	// create a service of type LoadBalancer with the annotation targeted by the ExternalDNS resource
+	t.Log("Creating source service")
+	service := defaultService(testService, testNamespace)
+	if err := kubeClient.Create(context.Background(), service); err != nil {
+		t.Fatalf("Failed to create test service %s/%s: %v", testNamespace, testService, err)
+	}
+	defer func() {
+		_ = kubeClient.Delete(context.TODO(), service)
+	}()
+
+	serviceIPs := make(map[string]struct{})
+	// get the IPs of the loadbalancer which is created for the service
+	if err := wait.PollImmediate(dnsPollingInterval, dnsPollingTimeout, func() (done bool, err error) {
+		t.Log("Getting IPs of service's load balancer")
+		var service corev1.Service
+		err = kubeClient.Get(context.TODO(), types.NamespacedName{
+			Namespace: testNamespace,
+			Name:      testService,
+		}, &service)
+		if err != nil {
+			return false, err
+		}
+
+		// if there is no associated loadbalancer then retry later
+		if len(service.Status.LoadBalancer.Ingress) < 1 {
+			return false, nil
+		}
+
+		// get the IPs of the loadbalancer
+		if service.Status.LoadBalancer.Ingress[0].IP != "" {
+			serviceIPs[service.Status.LoadBalancer.Ingress[0].IP] = struct{}{}
+		} else if service.Status.LoadBalancer.Ingress[0].Hostname != "" {
+			lbHostname := service.Status.LoadBalancer.Ingress[0].Hostname
+			ips, err := lookupARecord(lbHostname, googleDNSServer)
+			if err != nil {
+				t.Logf("Waiting for IP of loadbalancer %s", lbHostname)
+				// if the hostname cannot be resolved currently then retry later
+				return false, nil
+			}
+			for _, ip := range ips {
+				serviceIPs[ip] = struct{}{}
+			}
+		} else {
+			t.Logf("Waiting for loadbalancer details for service %s", testService)
+			return false, nil
+		}
+		t.Logf("Loadbalancer's IP(s): %v", serviceIPs)
+		return true, nil
+	}); err != nil {
+		t.Fatalf("Failed to get loadbalancer IPs for service %s/%s: %v", testNamespace, testService, err)
+	}
+
+	// try all nameservers and fail only if all failed
+	for _, nameSrv := range nameServers {
+		t.Logf("Looking for DNS record in nameserver: %s", nameSrv)
+		// verify that the IPs of the record created by ExternalDNS match the IPs of loadbalancer obtained in the previous step.
+		if err := wait.PollImmediate(dnsPollingInterval, dnsPollingTimeout, func() (done bool, err error) {
```

Comment:
As we discussed, this condition may easily overpass the whole e2e timeout of 1h because some providers return many name servers. Decreasing `dnsPollingTimeout` may help but it still leaves us with a lot of iteration which we need to go through entirely, which is a loss of time.

What I can propose here is to run the whole loop of the DNS check as a goroutine which would report failure or success into a channel. This would fork off a parallel check while the main test thread continue to the update of the secret with the right data. Then the main thread will just wait for the return from the channel (failure or success). This would help to remain with the single check as the other test cases.